src/java.desktop/share/classes/sun/awt/image/GifImageDecoder.java

```diff
@@ -396,7 +397,7 @@ private int sendPixels(int x, int y, int width, int height,
         int off = y * global_width + x2;
         boolean save = (curframe.disposal_method == GifFrame.DISPOSAL_SAVE);
         if (trans_pixel >= 0 && !curframe.initialframe) {
-            if (saved_image != null && model.equals(saved_model)) {
+            if (saved_image != null && model.equals(saved_model) && isSavedModelReliable) {
```

Comment:
Without the fix, if the model changes, we go the hard way, on a frame by frame basis.
With the fix, if the model changes, we go the hard way, for this and all subsequent frames.

But what was wrong with frame-by-frame, that "all subsequent frames" fixes ?

I suppose it must be that any frame that had a transparent pixel needs to be preserved for all subsequent frames, but I'm quite hazy on how that works.